From 4e4496bedf962830fd72f4d5f6a8153755b996fb Mon Sep 17 00:00:00 2001
From: Andreea Florescu <fandree@amazon.com>
Date: Tue, 6 Feb 2018 16:01:48 +0200
Subject: [PATCH 1/3] kvm_sys: fix alignment for x86 kvm structs

Add alignment for kvm_irq_routing, kvm_msrs and
kvm_coalesced_mmio_ring.

Signed-off-by: Andreea Florescu <fandree@amazon.com>
---
 kvm_sys/src/x86/bindings.rs | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/kvm_sys/src/x86/bindings.rs b/kvm_sys/src/x86/bindings.rs
index 40c970c..9b8542c 100644
--- a/kvm_sys/src/x86/bindings.rs
+++ b/kvm_sys/src/x86/bindings.rs
@@ -1273,6 +1273,7 @@ pub struct kvm_msrs {
     pub nmsrs: __u32,
     pub pad: __u32,
     pub entries: __IncompleteArrayField<kvm_msr_entry>,
+    __force_alignment: [u64; 0],
 }
 #[test]
 fn bindgen_test_layout_kvm_msrs() {
@@ -2759,6 +2760,7 @@ pub struct kvm_coalesced_mmio_ring {
     pub first: __u32,
     pub last: __u32,
     pub coalesced_mmio: __IncompleteArrayField<kvm_coalesced_mmio>,
+    __force_alignment: [u64; 0],
 }
 #[test]
 fn bindgen_test_layout_kvm_coalesced_mmio_ring() {
@@ -3608,6 +3610,7 @@ pub struct kvm_irq_routing {
     pub nr: __u32,
     pub flags: __u32,
     pub entries: __IncompleteArrayField<kvm_irq_routing_entry>,
+    __force_alignment: [u64; 0],
 }
 #[test]
 fn bindgen_test_layout_kvm_irq_routing() {
-- 
2.7.4

